### PR TITLE
Add comment regarding placement restrictions.

### LINKED ISSUE #11658: podAntiAffinity rule too broad in examples/filesystem.yaml
**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**

After adding a second filesystem to a three node cluster, ceph showed a health warning that the second mds daemon of one of the filesystems was missing.

It turned out that the rook operator had created the deployment, but the daemon could not be placed due to (`examples/filesystem.yaml`):

```
      podAntiAffinity:
        requiredDuringSchedulingIgnoredDuringExecution:
          - labelSelector:
              matchExpressions:
                - key: app
                  operator: In
                  values:
                    - rook-ceph-mds
```

I *do* understand that placing only one mds daemon on a node is a good approach if you have a large cluster with heavy loads. However, if you have the cluster mostly for HA and with few but powerful machines, it's no problem to have several mds on a machine, at least as long as they handle different filesystems.

**Expected behavior:**

It should be possible to have more than one filesystem without requiring twice the number of nodes.

So I came up with:

```
      podAntiAffinity:
        requiredDuringSchedulingIgnoredDuringExecution:
          - labelSelector:
              matchExpressions:
                - key: app
                  operator: In
                  values:
                    - rook-ceph-mds
                - key: rook_file_system
                  operator: In
                  values:
                    - myfs
```

Of course, it is annoying to have the name of the filesystem (again) in `values`, so maybe this should not be added to the example (or with a reference from `name` to the relevant part of the `podAntiAffinity` section). But at least the (possible) problem should be mentioned somewhere to warn the users and avoid surprises with more than one filesystem.

### PR DESCRIPTION
**Pull request by mnlipp**
**Description of your changes:**

Add comment regarding mds daemon placement to example file.

**Which issue is resolved by this Pull Request:**
Resolves #11658

### CODE REVIEW

File: deploy/examples/filesystem.yaml [hunk 1]
```diff
@@ -1,6 +1,8 @@
 #################################################################################################################
 # Create a filesystem with settings with replication enabled for a production environment.
-# A minimum of 3 OSDs on different nodes are required in this example.
+# A minimum of 3 OSDs on different nodes are required in this example. It allows only one
+# mds daemon per node, which may be too restrictive if you create more than one filesystem
```
Comment (travisn) [inline deploy/examples/filesystem.yaml:4]: What about simplifying to a single line, something like this? I like simple headers. :)
```suggestion
# If one mds daemon per node is too restrictive, see the podAntiAffinity below. 
```

**Comment by travisn** [comment]
The DCO action is failing, but I'm not clear why. The commit message may need a blank line before the Signed-off-by comment.

**Comment by travisn** [comment]
> The DCO action is failing, but I'm not clear why. The commit message may need a blank line before the Signed-off-by comment.

Don't worry about the DCO, it looks correct even though the bot isn't happy. But looks like the rewording on the header was lost.

**Comment by mnlipp** [comment]
> > The DCO action is failing, but I'm not clear why. The commit message may need a blank line before the Signed-off-by comment.
> 
> Don't worry about the DCO, it looks correct even though the bot isn't happy. But looks like the rewording on the header was lost.

Simply accepting it led to failing checks, because the corresponding commit message caused the checks to fail. And when I applied it as patch, the grammar was not correct, so I had to make an adjustment. Simply change it to your lking.

**Comment by travisn** [comment]
>Simply accepting it led to failing checks, because the corresponding commit message caused the checks to fail. And when I applied it as patch, the grammar was not correct, so I had to make an adjustment. Simply change it to your liking.

Accepting the commit from the github UI doesn't work well, as you're seeing. It should be simplest you just update the suggestion locally (manually) and then force push the single commit.

**Comment by mnlipp** [comment]
> Accepting the commit from the github UI doesn't work well, as you're seeing. It should be simplest you just update the suggestion locally (manually) and then force push the single commit.

That's what I did (after reverting the commit from the web UI). But this resulted in:

```
#################################################################################################################
# Create a filesystem with settings with replication enabled for a production environment.
# A minimum of 3 OSDs on different nodes are required in this example. It allows only one
# If one mds daemon per node is too restrictive, see the podAntiAffinity below. 
# (see podAntiAffinity below).
#  kubectl create -f filesystem.yaml
#################################################################################################################
```

"It allows only one" doesn't seem right, So I made it a complete senence.

**Comment by travisn** [comment]
> > Accepting the commit from the github UI doesn't work well, as you're seeing. It should be simplest you just update the suggestion locally (manually) and then force push the single commit.
> 
> That's what I did (after reverting the commit from the web UI). But this resulted in:
> 
> ```
> #################################################################################################################
> # Create a filesystem with settings with replication enabled for a production environment.
> # A minimum of 3 OSDs on different nodes are required in this example. It allows only one
> # If one mds daemon per node is too restrictive, see the podAntiAffinity below. 
> # (see podAntiAffinity below).
> #  kubectl create -f filesystem.yaml
> #################################################################################################################
> ```
> 
> "It allows only one" doesn't seem right, So I made it a complete senence.

My apologies for not being clear. I was suggesting this, which only adds a single line to the header.
```
# Create a filesystem with settings with replication enabled for a production environment.
# A minimum of 3 OSDs on different nodes are required in this example. 
# If one mds daemon per node is too restrictive, see the podAntiAffinity below. 
#  kubectl create -f filesystem.yaml
```

**Review by travisn — APPROVED** [review]

**Comment by travisn** [comment]
Thanks!

**Pull request merged**